lib/bin/lwm2m_carrier/lwm2m_carrier.rst

```diff
@@ -4,60 +4,124 @@ LwM2M carrier
 #############
 
 You can use the **LwM2M carrier** library in your nRF91 application in order to connect to the operator LwM2M network.
-Integrating this library into an application causes the device to connect to the device management platform autonomously.
+Integrating this library into an application causes the device connect to the operator's device management platform autonomously.
 
-Every released version is considered for certification with carriers.
-Refer to the :ref:`liblwm2m_carrier_changelog` to check the certification status of a particular release.
-Your final device must go through certification by the carrier.
+Examples of carrier device management platforms:
+
+* `Verizon's Thingspace <https://thingspace.verizon.com/>`_
+* `AT&T's IoT Platform <https://iotplatform.att.com/>`_
+
+This library uses LwM2M protocol to communicate with device manager platforms, but does not expose an LwM2M API to your application. Instead see :ref:`lwm2m_interface`.
 
 The :ref:`lwm2m_carrier` sample demonstrates how to run this library in an application.
+The Carrier library is also being used by the :ref:`asset_tracker`
 
-Prerequisites
-*************
-The LwM2M carrier library has dependencies on Zephyr and the |NCS|.
-See the :ref:`liblwm2m_carrier_changelog` of the library to check which version of |NCS| it is built for.
-To integrate the library into an application, the application must be based on a supported |NCS| version.
 
 Application integration
 ***********************
 To run the library in an application, you must implement the application with the API of the library.
 
+The :ref:`lwm2m_carrier` sample project configuration :file:`nrf/samples/nrf9160/lwm2m_carrier/prj.conf` contains all the configurations needed by the Carrier library.
+
+After calling :c:func:`lwm2m_carrier_init()`, your application can call the non-returning function :c:func:`lwm2m_carrier_run`.
+
+:c:func:`lwm2m_carrier_event_handler` must be implemented by your application. This is shown in the :ref:`lwm2m_carrier` sample.
+
+See :file:`lwm2m_carrier.h` for all the events and API.
+
+Carrier Events
+==============
+:c:macro:`LWM2M_CARRIER_EVENT_BSDLIB_INIT`
+   :ref:`bsdlib` Is done initializing and can now be used. (See Requirements and Application limitations).
+
+:c:macro:`LWM2M_CARRIER_EVENT_CONNECTING`, :c:macro:`LWM2M_CARRIER_EVENT_CONNECTED`, :c:macro:`LWM2M_CARRIER_EVENT_DISCONNECTING`, :c:macro:`LWM2M_CARRIER_EVENT_DISCONNECTED`
+   Connecting or disconnecting from the LTE network. Will appear during the bootstrapping process and startup.
```

Comment:
I think `LWM2M_CARRIER_EVENT_DISCONNECTING` and `LWM2M_CARRIER_EVENT_DISCONNECTED` can also sometimes appear during FOTA when erasing the firmware if the image is dirty, but I'm not sure if that needs to be stated.